File: python/AbstractFactory/complete/LegalDocumentFactory.py
from DocumentFactory import DocumentFactory
from LegalDocuments import *

class LegalDocumentFactory(DocumentFactory):

    def create_resume(self, content):
        '''
        returns a new LegalResume object with the content parameter
        '''
        return LegalResume(content)

    def create_letter(self, content):
        '''
        returns a new LegalLetter object with the content parameter
        '''
        return LegalLetter(content)

File: java/AbstractFactory/complete/LegalDocumentFactory.java
class LegalDocumentFactory extends DocumentFactory{

  public Resume create_resume(String content){
    //returns a new LegalResume object with the content parameter
    return new LegalResume(content);
  }

  public Letter create_letter(String content){
    //returns a new LegalLetter object with the content parameter
    return new LegalLetter(content);
  }
}
